src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineStartService.java

```diff
@@ -0,0 +1,143 @@
+	 */
+	public Long start(UUID id, LaunchRequest request, Locale locale)
+			throws IridaWorkflowNotFoundException, ReferenceFileRequiredException {
+		IridaWorkflow workflow = workflowsService.getIridaWorkflow(id);
+		IridaWorkflowDescription description = workflow.getWorkflowDescription();
+
+		/*
+		PARAMETERS
+		 */
+		IridaWorkflowNamedParameters namedParameters = null;
+
+		if (request.getSavedParameters() != null) {
+			namedParameters = namedParametersService.read(request.getSavedParameters());
+		}
+
+		// Make sure there is a reference files if one is required.
+		if (description.requiresReference() && request.getReference() == null) {
+			throw new ReferenceFileRequiredException(
+					messageSource.getMessage("server.ReferenceFiles.notFound", new Object[] {}, locale));
+		}
+
+		if (request.getAutomatedProjectId() != null) {
+			Project project = projectService.read(request.getAutomatedProjectId());
+			String statusMessage = messageSource.getMessage("analysis.template.status.new", new Object[] {}, locale);
+			AnalysisSubmissionTemplate template = submissionService.createSingleSampleSubmissionTemplate(workflow,
+					request.getReference(), request.getParameters(), namedParameters, request.getName(), statusMessage,
+					request.getDescription(), project, request.isUpdateSamples(), request.sendEmailOnCompletion(),
+					request.sendEmailOnError());
+			return template.getId();
+		} else {
+			/*
+			SHARE RESULTS BACK TO PROJECTS?
+		 	*/
+			List<Project> projects = new ArrayList<>();
+			if (request.getProjects()
+					.size() > 0) {
+				projects = (List<Project>) projectService.readMultiple(request.getProjects());
+			}
+
+			/*
+			SEQUENCE FILES
+		 	*/
+			List<SingleEndSequenceFile> singles = new ArrayList<>();
+			List<SequenceFilePair> pairs = new ArrayList<>();
+			// Check for single ended sequence files
+			Iterable<SequencingObject> sequencingObjects = sequencingObjectService.readMultiple(request.getFileIds());
+			if (description.acceptsSingleSequenceFiles()) {
+				sequencingObjects.forEach(sequencingObject -> {
+					if (sequencingObject instanceof SingleEndSequenceFile) {
+						singles.add((SingleEndSequenceFile) sequencingObject);
+					}
+				});
+			} else if (description.acceptsPairedSequenceFiles()) {
+				sequencingObjects.forEach(sequencingObject -> {
+					if (sequencingObject instanceof SequenceFilePair) {
+						pairs.add((SequenceFilePair) sequencingObject);
+					}
+				});
+			}
+
+			IridaWorkflowInput inputs = description.getInputs();
+			cartService.emptyCart();
+			if (inputs.requiresSingleSample()) {
+				submissionService.createSingleSampleSubmission(workflow, request.getReference(), singles, pairs,
+						request.getParameters(), namedParameters, request.getName(), request.getDescription(), projects,
+						request.isUpdateSamples(), request.sendEmailOnCompletion(), request.sendEmailOnError());
+				return -1L;
```

Comment:
Sorry I may have missed something but why are we returning a -1 here?

Reply:
I updated with a comment in 18c7a1d063ddf1dcc10da9ed1b721ba5fa100472.

This is something that Tom and I set up yesterday.  It is a flag for the UI.  If the success response returns a value above -1 then it knows that it is the identifier for the new submission.  In this case since multiple submissions are created it does not makes sense to go to a specific one, so this flag tells the UI to just go the the listing page.

Reply:
Ah that makes sense. Thanks!